app/javascript/components/ui/ProductCard.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from "react";
+
+import { classNames } from "$app/utils/classNames";
+
+type BaseProps = {
+  children: React.ReactNode;
+  className?: string;
+} & React.HTMLAttributes<HTMLElement>;
+
+export const ProductCard = ({ children, className, ...props }: BaseProps) => (
+  <article
+    className={classNames(
+      "relative grid grid-rows-[auto_1fr_auto] rounded border border-border bg-background transition-all duration-150 hover:shadow",
```

Comment:
Can we use `flex` and set `flex-1` on the header instead of these grid rows? (would be nice if this was less tied to the number/type of children)